core/utils/utils.go

```diff
@@ -818,13 +818,23 @@ type CronTicker struct {
 	ch chan time.Time
 }
 
-func NewCronTicker(schedule string) (CronTicker, error) {
+func NewCronTicker(schedule string, jitterWithin time.Duration) (CronTicker, error) {
 	cron := cron.New(cron.WithSeconds())
 	ch := make(chan time.Time, 1)
 	_, err := cron.AddFunc(schedule, func() {
-		select {
-		case ch <- time.Now():
-		default:
+		if jitterWithin > 0 {
+			delay := time.Duration(mrand.Int63n(int64(jitterWithin)))
```

Comment:
I would expect the jitter to be +/-, not a delay

Reply:
hm maybe we could call it something else in this case, what do you think? Subtracting from the Cron-decided time would be a bit tricky